### PR TITLE
Refactor HTML doc generation

### PR DESCRIPTION
**Pull request by kianmeng**
This PR refactors the HTML doc generation which includes:

- Use common source URL
- Add link back to Github URL for function header.
- Add Changelog to module info page and HTML doc.
- Badges and more badges.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -1,16 +1,18 @@
 defmodule ExSecp256k1.MixProject do
   use Mix.Project
 
+  @source_url "https://github.com/ayrat555/ex_secp256k1"
```
Comment (ayrat555) [inline mix.exs:4]: @kianmeng can you please set it to `https://github.com/omgnetwork/ex_secp256k1`?

Comment (kianmeng) [reply mix.exs]: > @kianmeng can you please set it to `https://github.com/omgnetwork/ex_secp256k1`?

@ayrat555 Done.

File: README.md [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+Rust Nif that wraps a couple functions from the
+[libsecp256k1](https://github.com/paritytech/libsecp256k1) Rust library. It
+only wraps `secp256k1` function used in Ethereum.
 
 ## Installation
 
 `ex_secp256k1` requires Rust to be installed.
 
-The package can be installed by adding `ex_secp256k1` to your list of dependencies in `mix.exs`:
+The package can be installed by adding `ex_secp256k1` to your list of
```
Comment (ayrat555) [inline README.md:11]: I don't think it has to be split into multiple lines. It looks ok https://hexdocs.pm/ex_secp256k1/readme.html. the same comment for line 3, 22, 27, 52

Comment (kianmeng) [reply README.md:15]: > I don't think it has to be split into multiple lines. It looks ok https://hexdocs.pm/ex_secp256k1/readme.html. the same comment for line 3, 22, 27, 52

@ayrat555 Done.

File: README.md [hunk 3]
```diff
@@ -1,12 +1,19 @@
 # ExSecp256k1
 
-Rust Nif that wraps a couple functions from the [libsecp256k1](https://github.com/paritytech/libsecp256k1) rust library. It only wraps secp256k1 functions used in Ethereum.
+[![Hex.pm](https://img.shields.io/hexpm/v/ex_secp256k1.svg)](https://hex.pm/packages/ex_secp256k1)
```
Comment (ayrat555) [inline README.md:3]: 👍

File: README.md [hunk 4]
```diff
@@ -1,12 +1,19 @@
...
+[![Hex.pm](https://img.shields.io/hexpm/dt/ex_secp256k1.svg)](https://hex.pm/packages/ex_secp256k1)
+[![Hex.pm](https://img.shields.io/hexpm/l/ex_secp256k1.svg)](https://hex.pm/packages/ex_secp256k1)
+[![Github.com](https://img.shields.io/github/last-commit/omgnetwork/ex_secp256k1.svg)](https://github.com/omgnetwork/ex_secp256k1)
+
+
+Rust Native Implemented Function (NIF) that wraps a couple functions from the [libsecp256k1](https://github.com/paritytech/libsecp256k1) Rust library. It only wraps `secp256k1` function used in Ethereum.
```
Comment (ayrat555) [inline README.md:9]:
```suggestion
Rust Native Implemented Function (NIF) that wraps a couple functions from the [libsecp256k1](https://github.com/paritytech/libsecp256k1) Rust library. It only wraps `secp256k1` functions used in Ethereum.
```

**Review by ayrat555 — COMMENTED** [review]
the last comment

**Review by ayrat555 — APPROVED** [review]

**Comment by ayrat555** [comment]
@kianmeng  Thanks for your contribution!

**Pull request merged**